### PR TITLE
tmf: fix invalid 1ns histogram span

### PR DESCRIPTION
**Pull request by vladarama**
Fixes: #47 

This PR fixes an issue where histograms would be allowed to have a 1ns window span, which should not be possible.

Signed-off-by: Vlad Arama <vlad.arama@ericsson.com>

### CODE REVIEW

File: tmf/org.eclipse.tracecompass.tmf.ui/src/org/eclipse/tracecompass/tmf/ui/views/histogram/HistogramView.java [hunk 1]
```diff
@@ -610,6 +610,11 @@ private void setNewRange(long startTime, long duration) {
             realStart = fTraceStartTime;
         }
 
+        if (duration == 1) {
+            fTimeSpanControl.setValue(fWindowSpan);
```
Comment (MatthewKhouzam) [inline tmf/org.eclipse.tracecompass.tmf.ui/src/org/eclipse/tracecompass/tmf/ui/views/histogram/HistogramView.java:614]: Do we need this?

File: tmf/org.eclipse.tracecompass.tmf.ui/src/org/eclipse/tracecompass/tmf/ui/views/histogram/HistogramView.java [hunk 2]
```diff
@@ -610,6 +610,11 @@ private void setNewRange(long startTime, long duration) {
             realStart = fTraceStartTime;
         }
 
+        if (duration == 1) {
```
Comment (MatthewKhouzam) [inline tmf/org.eclipse.tracecompass.tmf.ui/src/org/eclipse/tracecompass/tmf/ui/views/histogram/HistogramView.java:613]: let's do <= 1? that way we can catch 0 too?

**Review by MatthewKhouzam — CHANGES_REQUESTED** [review]
Test with negative time ranges for giggles?

File: tmf/org.eclipse.tracecompass.tmf.ui/src/org/eclipse/tracecompass/tmf/ui/views/histogram/HistogramView.java (hunk 2 above)
Comment (vladarama) [reply tmf/org.eclipse.tracecompass.tmf.ui/src/org/eclipse/tracecompass/tmf/ui/views/histogram/HistogramView.java]: Good idea 👍

File: tmf/org.eclipse.tracecompass.tmf.ui/src/org/eclipse/tracecompass/tmf/ui/views/histogram/HistogramView.java (hunk 1 above)
Comment (vladarama) [reply tmf/org.eclipse.tracecompass.tmf.ui/src/org/eclipse/tracecompass/tmf/ui/views/histogram/HistogramView.java:614]: This sets the value in the text box to its previous value. 
I.e. if we don't use that, we will still have 1ns still written in the text box even if the histogram isn't updated.

Comment (MatthewKhouzam) [reply tmf/org.eclipse.tracecompass.tmf.ui/src/org/eclipse/tracecompass/tmf/ui/views/histogram/HistogramView.java:614]: Indeed, I misread duration... oops :)

**Review by MatthewKhouzam — APPROVED** [review]
Good w/ me!

**Review by bhufmann — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**